src/db/write.rs

```diff
@@ -338,15 +344,39 @@ impl<'a> Write<'a> {
 
     // Return value is the hash of the new commit.
     #[allow(clippy::too_many_arguments)]
-    pub async fn commit(mut self, head_name: &str) -> Result<String, CommitError> {
+    pub async fn commit(
+        mut self,
+        head_name: &str,
+        generate_diffs: bool,
+    ) -> Result<(String, BTreeMap<String, Vec<String>>), CommitError> {
```

Comment:
OOC, why BTreeMap vs HashMap?

Reply:
I thought I answered this already. I thought that I might take advantage of ordered sets/vectors at some point. Never did that. Switching to HashMap for now.